### PR TITLE
Add gameplay modifier 'Play as Opponent'

### PR DESCRIPTION
**Pull request by Gui-iago**
hopefully it looks good

### CODE REVIEW

File: source/PlayState.hx [hunk 1]
```diff
@@ -3973,13 +3988,18 @@ class PlayState extends MusicBeatState
 					//	boyfriend.holdTimer = 0;
 					//}
 				//}else{
+					if(opponentChart) {
+						dad.playAnim('sing' + animToPlay + daAlt, true);
+						dad.holdTimer = 0;
+					}
```
Comment (l1ttleO) [inline source/PlayState.hx:3994]: create a `char` variable and use it (like in `opponentNoteHit`)

File: source/PlayState.hx [hunk 2]
```diff
@@ -3879,12 +3889,17 @@ class PlayState extends MusicBeatState
 			if(SONG.notes[curSection].gfSection || note.noteType == 'GF Sing') {
 				char = gf;
 			}
-
+			if(opponentChart) {
+				boyfriend.playAnim('sing' + animToPlay, true);
+				boyfriend.holdTimer = 0;
+			}
+			if(!opponentChart) {
```
Comment (l1ttleO) [inline source/PlayState.hx:3896]:
```suggestion
			else {
```

File: source/PlayState.hx [hunk 3]
```diff
@@ -1767,10 +1769,15 @@ class PlayState extends MusicBeatState
...
-					gottaHitNote = !section.mustHitSection;
-				}
+				if (songNotes[1] > 3 && !opponentChart)
+					{
+						gottaHitNote = !section.mustHitSection;
+					}
+					else if (songNotes[1] <= 3 && opponentChart)
+					{
+						gottaHitNote = !section.mustHitSection;
+					}
```
Comment (l1ttleO) [inline source/PlayState.hx:1779]:
```suggestion
				{
					gottaHitNote = !section.mustHitSection;
				}
				else if (songNotes[1] <= 3 && opponentChart)
				{
					gottaHitNote = !section.mustHitSection;
				}
```
indenting?

File: source/PlayState.hx [hunk 4]
```diff
@@ -3765,6 +3774,7 @@ class PlayState extends MusicBeatState
 		if(daNote.gfNote) {
 			char = gf;
 		}
+		if (opponentChart) char = dad;
```
Comment (l1ttleO) [inline source/PlayState.hx:3777]: should put this before `if(daNote.gfNote)`

File: source/PlayState.hx [hunk 5]
```diff
@@ -3879,12 +3889,17 @@ class PlayState extends MusicBeatState
...
+				boyfriend.playAnim('sing' + animToPlay, true);
+				boyfriend.holdTimer = 0;
+			}
+			if(!opponentChart) {
 			char.playAnim('sing' + animToPlay, true);
 			char.holdTimer = 0;
 			if (ClientPrefs.moveCameraInNoteDirection)
 				moveCamera(true, animToPlay);
 		}
+	}
```
Comment (l1ttleO) [inline source/PlayState.hx:3902]:
```suggestion
				char.playAnim('sing' + animToPlay, true);
				char.holdTimer = 0;
				if (ClientPrefs.moveCameraInNoteDirection)
					moveCamera(true, animToPlay);
			}
		}
```
indenting??

File: source/PlayState.hx [hunk 6]
```diff
@@ -3830,7 +3840,7 @@ class PlayState extends MusicBeatState
 				boyfriend.stunned = false;
 			});
 
-			if(boyfriend.hasMissAnimations && ClientPrefs.playMissAnimations) {
+			if(boyfriend.hasMissAnimations && !opponentChart && ClientPrefs.playMissAnimations) {
 				boyfriend.playAnim('sing' + singAnimations[Std.int(Math.abs(direction))] + 'miss', true);
```
Comment (l1ttleO) [inline source/PlayState.hx:3844]: create a `char` variable and use it (like in `opponentNoteHit` but without gf)

**Review by l1ttleO — CHANGES_REQUESTED** [review]

**Comment by Gui-iago** [comment]
I did a bunch of mistakes at the end there, but in general it works without any problems, I couldn't really figure out how the health bar worked for PFNF, so I didn't do anything, and reverted back to it's original state, I'm still learning how to code for FNF in general, so hopefully this looks good and someone can fix my dumb mistakes

File: source/PlayState.hx [hunk 7]
```diff
@@ -1948,7 +1956,8 @@ class PlayState extends MusicBeatState
 						babyArrow.x += FlxG.width / 2 + 25;
 					}
 				}
-				opponentStrums.add(babyArrow);
+				if (!opponentChart || opponentChart && ClientPrefs.middleScroll) opponentStrums.add(babyArrow);
```
Comment (l1ttleO) [inline source/PlayState.hx:1959]:
```suggestion
				if (!opponentChart || ClientPrefs.middleScroll) opponentStrums.add(babyArrow);
```

File: source/PlayState.hx [hunk 8]
```diff
@@ -1937,7 +1944,8 @@ class PlayState extends MusicBeatState
 
 			if (player == 1)
 			{
-				playerStrums.add(babyArrow);
+				if (!opponentChart || opponentChart && ClientPrefs.middleScroll) playerStrums.add(babyArrow);
```
Comment (l1ttleO) [inline source/PlayState.hx:1947]:
```suggestion
				if (!opponentChart || ClientPrefs.middleScroll) playerStrums.add(babyArrow);
```

File: source/PlayState.hx [hunk 9]
```diff
@@ -1767,10 +1769,15 @@ class PlayState extends MusicBeatState
 
 				var gottaHitNote:Bool = section.mustHitSection;
 
-				if (songNotes[1] > 3)
+				if (songNotes[1] > 3 && !opponentChart)
```
Comment (l1ttleO) [inline source/PlayState.hx:1772]:
```suggestion
				if (songNotes[1] > 3)
```

File: source/PlayState.hx [hunk 10]
```diff
@@ -1767,10 +1769,15 @@ class PlayState extends MusicBeatState
...
 				{
 					gottaHitNote = !section.mustHitSection;
 				}
+				else if (songNotes[1] <= 3 && opponentChart)
```
Comment (l1ttleO) [inline source/PlayState.hx:1776]:
```suggestion
				else if (opponentChart)
```

File: source/PlayState.hx [hunk 11]
```diff
@@ -3976,13 +3991,18 @@ class PlayState extends MusicBeatState
 					//	boyfriend.holdTimer = 0;
 					//}
 				//}else{
+					if(opponentChart && !note.gfNote) {
```
Comment (l1ttleO) [inline source/PlayState.hx:3994]:
```suggestion
					if(opponentChart) {
```

File: source/PlayState.hx [hunk 12]
```diff
@@ -3976,13 +3991,18 @@ class PlayState extends MusicBeatState
...
+						dad.playAnim('sing' + animToPlay + daAlt, true);
+						dad.holdTimer = 0;
+					}
 					if(note.gfNote) {
```
Comment (l1ttleO) [inline source/PlayState.hx:3998]:
```suggestion
					else if(note.gfNote) {
```

File: source/PlayState.hx [hunk 13]
```diff
@@ -3976,13 +3991,18 @@ class PlayState extends MusicBeatState
...
 						gf.playAnim('sing' + animToPlay + daAlt, true);
 						gf.holdTimer = 0;
 					} else {
 						boyfriend.playAnim('sing' + animToPlay + daAlt, true);
 						boyfriend.holdTimer = 0;
 					}
+
```
Comment (l1ttleO) [inline source/PlayState.hx:4005]:
```suggestion
```

File: source/PlayState.hx [hunk 14]
```diff
@@ -3882,11 +3892,16 @@ class PlayState extends MusicBeatState
...
 				char = gf;
 			}
-
-			char.playAnim('sing' + animToPlay, true);
-			char.holdTimer = 0;
-			if (ClientPrefs.moveCameraInNoteDirection)
-				moveCamera(true, animToPlay);
+			if(opponentChart) {
+				boyfriend.playAnim('sing' + animToPlay, true);
+				boyfriend.holdTimer = 0;
```
Comment (l1ttleO) [inline source/PlayState.hx:3897]:
```suggestion
				boyfriend.holdTimer = 0;
				if (ClientPrefs.moveCameraInNoteDirection)
					moveCamera(false, animToPlay);
```

**Review by l1ttleO — CHANGES_REQUESTED** [review]
Also delete these files
image

**Review by l1ttleO — APPROVED** [review]
Thanks for your contribution

**Pull request merged**